src/search-insights.ts

```diff
@@ -49,8 +49,21 @@ export function activate(context: sourcegraph.ExtensionContext): void {
         distinctUntilChanged((a, b) => isEqual(a, b))
     )
 
+    // Further in main context.subscriptions we have nested calls of context.subscriptions
+    // as well, because of this we have to manage inner subscription ourselves. This
+    // subscription bag needed to unsubscribe nested calls in a moment when user/org settings
+    // have been updated.
+    let previousSubscriptions = new Subscription()
+
     context.subscriptions.add(
-        insightChanges.pipe(mergeAll()).subscribe(([id, insight]) => {
+        insightChanges.pipe(
+            tap(() => {
+                previousSubscriptions.unsubscribe()
+                previousSubscriptions = new Subscription()
```

Comment:
I wonder if this line (instantiating a new instance) is even necessary 🤔

Reply:
Yes, because if you unsubscribed once then a next subscriptions will be immediately unsubscribed as well

Reply:
Oh right